jina/orchestrate/flow/base.py

```diff
@@ -2401,3 +2410,14 @@ def __getattribute__(self, item):
             )
 
         return obj
+
+    @property
+    def _flow_id(self) -> str:
+        import hashlib
+
+        if self._build_level.value < FlowBuildLevel.GRAPH.value:
+            self.build(copy_flow=False)
+
+        return hashlib.sha256(
+            json.dumps(self._get_graph_representation()).encode('utf-8')
```

Comment:
but is the most representing thing. Other args I consider them less important. Which args would u add?

Reply:
yes but it might not be enough. I'd say it would be great if we manage to get executor names, otherwise this can generate false metrics.
check code snippet for reference
```python
import hashlib
import json

from jina import Flow

flow1 = Flow().add(uses='jinahub://DummyHubExecutor').add(needs='gateway')
flow2 = Flow().add().add(needs='gateway')


with flow1, flow2:
    print(hashlib.sha256(
        json.dumps(flow1._get_graph_representation()).encode('utf-8')).hexdigest())
    print(hashlib.sha256(
        json.dumps(flow2._get_graph_representation()).encode('utf-8')).hexdigest())

```

Reply:
Executor names are not available at the `gateway` level

Reply:
But you are right is too poor. I will make it a draft